sender-stackdriver/src/main/java/zipkin2/reporter/stackdriver/internal/AwaitableUnaryClientCallListener.java

```diff
@@ -28,16 +30,22 @@
 
   Object result; // guarded by this
 
+  @VisibleForTesting
+  static long TIMEOUT_MS = 5000; // how long to wait for server response in milliseconds
+
   /**
    * Blocks until {@link #onClose}. Throws if no value was received, multiple
-   * values were received, or there was a status error.
+   * values were received, there was a status error, or waited longer than {@link #TIMEOUT_MS}.
    */
   V await() throws IOException {
     boolean interrupted = false;
     try {
       while (true) {
         try {
-          countDown.await();
+          if (!countDown.await(this.TIMEOUT_MS, TimeUnit.MILLISECONDS)) {
+            throw new IllegalStateException("timeout waiting for onClose. TIMEOUT_MS=" + TIMEOUT_MS
```

Comment:
no need to upper underscore the message. if this becomes configurable later we would change the log message without a good reason. upper underscore in the message is distracting imho